src/borg/archiver.py

```diff
@@ -2285,12 +2285,16 @@ def build_parser(self, prog=None):
 
     @staticmethod
     def add_archives_filters_args(subparser):
+        filters_group = subparser.add_argument_group('filters', 'Archive filters can be applied on repository targets.')
+        filters_group.add_argument('-P', '--prefix', dest='prefix', type=prefix_spec, default='',
+                               help='only consider archive names starting with this prefix')
```

Comment:
with `args.prefix == None` [this list comprehension](https://github.com/borgbackup/borg/pull/1554/files#diff-d3eb25b2d47fdf880fd23aae789dd9efR150) would break.

no, it isn't dangerous to use delete without --prefix argument, that is actually what you want when you use one of the three filter arguments. the 'filter code' wouldn't be invoked is only invoked when at least one of these is used.
atm i can't think of an undesired outcome. do you have an example?